app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt

```diff
@@ -128,7 +128,7 @@ class BrowserFragment : EngineViewLifecycleFragment(), Session.Observer {
             NavigationEvent.SETTINGS -> ScreenController.showSettingsScreen(fragmentManager!!)
             NavigationEvent.LOAD_URL -> {
                 (activity as MainActivity).onTextInputUrlEntered(value!!, autocompleteResult!!, UrlTextInputLocation.MENU)
-                setOverlayVisible(false)
+                if (value != Urls.APP_HOME) setOverlayVisible(false)
```

Comment:
Rather than stop this at the LOAD_URL layer, can we stop it at the input validation layer? i.e. in the commit listener, don't call load url if the URL is invalid? This seems less reactive to me and doesn't require us to try to undo state (e.g. we're recording a telemetry event for this page load).

Reply:
Good point.